common/src/main/java/whocraft/tardis_refined/common/tardis/CorridorGenerator.java

```diff
@@ -0,0 +1,261 @@
+            }
+
+            return createResourceLocation( "room_wall_nsw");
+        }
+
+
+
+
+        return new ResourceLocation("");
+    }
+
+    private static boolean doesBlockExist(BlockState blockState) {
+        return !blockState.isAir();
+    }
+
+    private static boolean isBlockRoomConnection(BlockState blockState) {
+        return blockState == Blocks.REDSTONE_BLOCK.defaultBlockState();
+    }
+
+
+    private static ResourceLocation getCorridorPieceResourceLocation(BlockState[] results) {
+
+        // CHECK IF NORTH/SOUTH/EAST/WEST
+        if (doesBlockExist(results[0]) && doesBlockExist(results[1]) && !doesBlockExist(results[2]) && !doesBlockExist(results[3])) {
+            return createResourceLocation( "corridor_piece_ns");
+        }
+
+        if (!doesBlockExist(results[0]) && !doesBlockExist(results[1]) && doesBlockExist(results[2]) && doesBlockExist(results[3])) {
+            return createResourceLocation( "corridor_piece_ew");
+        }
+
+        if (doesBlockExist(results[0]) && doesBlockExist(results[1]) && doesBlockExist(results[2]) && doesBlockExist(results[3])) {
+            return createResourceLocation( "corridor_piece_cross");
+        }
+
+        // CHECK IF NORTH/SOUTH/EAST/WEST
+        if (doesBlockExist(results[0]) && !doesBlockExist(results[1]) && doesBlockExist(results[2]) && !doesBlockExist(results[3])) {
+            return createResourceLocation( "corridor_piece_ne");
+        }
+
+        if (doesBlockExist(results[0]) && !doesBlockExist(results[1]) && !doesBlockExist(results[2]) && doesBlockExist(results[3])) {
+            return createResourceLocation( "corridor_piece_nw");
+        }
+
+        if (!doesBlockExist(results[0]) && doesBlockExist(results[1]) && doesBlockExist(results[2]) && !doesBlockExist(results[3])) {
+            return createResourceLocation( "corridor_piece_se");
+        }
+
+        if (!doesBlockExist(results[0]) && doesBlockExist(results[1]) && !doesBlockExist(results[2]) && doesBlockExist(results[3])) {
+            return createResourceLocation( "corridor_piece_sw");
+        }
+
+        if (doesBlockExist(results[0]) && !doesBlockExist(results[1]) && doesBlockExist(results[2]) && doesBlockExist(results[3])) {
+            return createResourceLocation( "corridor_piece_new");
+        }
+
+        if (doesBlockExist(results[0]) && doesBlockExist(results[1]) && doesBlockExist(results[2]) && !doesBlockExist(results[3])) {
+            return createResourceLocation( "corridor_piece_nse");
+        }
+
+        if (!doesBlockExist(results[0]) && doesBlockExist(results[1]) && doesBlockExist(results[2]) && doesBlockExist(results[3])) {
+            return createResourceLocation( "corridor_piece_sew");
+        }
+
+        if (doesBlockExist(results[0]) && doesBlockExist(results[1]) && !doesBlockExist(results[2]) && doesBlockExist(results[3])) {
+            return createResourceLocation( "corridor_piece_nsw");
+        }
+
+        return new ResourceLocation("");
```

Comment:
Have we tested what happens if this this condition occurs? If so, are we handling empty resource locations properly?

Reply:
Yes! It places nothing. No issues with it.